test/functional/discovered_hosts_controller_test.rb

```diff
@@ -84,27 +84,27 @@ def test_edit_form_build_mode_enabled
   end
 
   def test_edit_form_quick_submit
-    disable_taxonomies do
-      domain = FactoryBot.create(:domain)
-      hostgroup = FactoryBot.create(:hostgroup, :with_subnet, :with_environment, :with_rootpass, :with_os, :domain => domain)
-      new_ip = hostgroup.subnet.ipaddr
-      host = discover_host_from_facts(@facts.merge({
-                                                     'ipaddress' => new_ip,
-                                                     'ipaddress_eth0' => new_ip
-                                                   }))
-      get :edit, params: {
-        :id => host.id,
-        :quick_submit => true,
-        :host => {
-          :hostgroup_id => hostgroup.id
-        } }, session: set_session_user_default_manager
-
-      managed_host = Host.find(host.id)
-      assert managed_host.build
-      assert_redirected_to host_url(managed_host)
-      assert_equal hostgroup.id, managed_host.hostgroup_id
-      assert_match(/Successfully/, flash[:success])
-    end
+    organization = FactoryBot.create(:organization)
+    location = FactoryBot.create(:location)
+    domain = FactoryBot.create(:domain, :organizations => [organization], :locations => [location] )
```

Comment:
You don't need to manually create the taxomonies, the domain factory assigns them by default - https://github.com/theforeman/foreman/blob/develop/test/factories/domain.rb#L9 and so does the hostgroup factory, so if you assign different ones here you may get wrong result

Reply:
good catch, thanks! rebased with the corrections.